src/artemis/external_interaction/ispyb/store_in_ispyb.py

```diff
@@ -233,77 +220,234 @@ def _store_data_collection_table(self, data_collection_group_id: int) -> int:
             "file_template"
         ] = f"{self.detector_params.prefix}_{self.run_number}_master.h5"
 
-        return self.mx_acquisition.upsert_data_collection(list(params.values()))
+        return mx_acquisition.upsert_data_collection(list(params.values()))
 
-    def _store_position_table(self, dc_id: int) -> int:
-        params = self.mx_acquisition.get_dc_position_params()
 
-        params["id"] = dc_id
+class StoreRotationInIspyb(StoreInIspyb):
+    ispyb_params: RotationIspybParams
+    data_collection_id: int | None = None
+
+    def __init__(self, ispyb_config, parameters: RotationInternalParameters) -> None:
+        self.full_params: RotationInternalParameters = parameters
+        self.ispyb_params: RotationInternalParameters = (
+            parameters.artemis_params.ispyb_params
+        )
+        self.detector_params = parameters.artemis_params.detector_params
+        self.omega_start = self.detector_params.omega_start
+
+        if self.ispyb_params.xtal_snapshots_omega_start:
+            self.xtal_snapshots = self.ispyb_params.xtal_snapshots_omega_start
+        else:
+            self.xtal_snapshots = []
+            LOGGER.warning("No xtal snapshot paths sent to ISPyB!")
+        self.experiment_type = "SAD"
+        super().__init__(ispyb_config, parameters)
+
+    def _mutate_datacollection_params_for_experiment(
+        self, params: dict[str, Any]
+    ) -> dict[str, Any]:
+        assert self.full_params is not None
+        params["axis_range"] = self.full_params.experiment_params.rotation_angle
+        params["axis_end"] = (
+            self.full_params.experiment_params.omega_start
+            + self.full_params.experiment_params.rotation_angle
+        )
+        params["n_images"] = self.full_params.experiment_params.get_num_images()
+        return params
+
+    def _store_scan_data(self, conn: Connector):
+        data_collection_group_id = self._store_data_collection_group_table(conn)
+        self.data_collection_group_id = data_collection_group_id
+        data_collection_id = self._store_data_collection_table(
+            conn, data_collection_group_id
+        )
+        self.data_collection_id = data_collection_id
+        self._store_position_table(conn, data_collection_id)
+
+        return data_collection_id, data_collection_group_id
+
+    def begin_deposition(self):
+        with ispyb.open(self.ISPYB_CONFIG_PATH) as conn:
+            return self._store_scan_data(conn)
+
+    def end_deposition(self, success: str, reason: str):
```

Comment:
Should: Is the only difference between this and the grid one how many dcids we have? Feels like we can move some more logic into the parent here